File: backend/app/models/topic.py
"""
话题表 - Sprint 3.5 运营功能
"""
from datetime import datetime
from sqlalchemy import Column, String, Text, Boolean, DateTime, Integer

from .base import Base
from .user import gen_uuid


class Topic(Base):
    __tablename__ = "topics"

    id = Column(String(36), primary_key=True, default=gen_uuid)
    name = Column(String(50), nullable=False, comment="话题名称")
    description = Column(Text, nullable=True, comment="话题描述")
    cover_image = Column(String(500), nullable=True, comment="封面图片URL")
    post_count = Column(Integer, default=0, nullable=False, comment="关联帖子数")
    is_active = Column(Boolean, default=True, nullable=False, comment="是否启用")
    sort_order = Column(Integer, default=0, nullable=False, comment="排序权重")
    created_at = Column(DateTime, default=datetime.utcnow)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

File: miniapp/src/api/topic.ts
/**
 * 话题相关 API
 */
import request from '@/utils/request'

export interface Topic {
  id: string
  name: string
  description: string | null
  cover_image: string | null
  post_count: number
  is_active: boolean
  sort_order: number
  created_at: string
  updated_at: string | null
}

/**
 * 获取启用的话题列表（公开接口）
 */
export async function getActiveTopics(): Promise<Topic[]> {
  const res: any = await request({
    url: '/api/v1/topics/active',
    method: 'GET',
  })
  // request 工具会自动展开响应，直接返回 data
  // 所以这里需要检查 res 是否是数组
  return Array.isArray(res) ? res : []
}
